HW2/P4/driver.py

```diff
@@ -15,17 +15,49 @@
 from timer import Timer
 import threading
 
+# Worked with Abhishek Malali 
 def py_median_3x3(image, iterations=10, num_threads=1):
-    ''' repeatedly filter with a 3x3 median '''
     tmpA = image.copy()
     tmpB = np.empty_like(tmpA)
+    # initialize events
+    events = []
+    for i in range(num_threads):
+        tmp = []
+        for j in range(iterations):
+            tmp.append(threading.Event())
+        events.append(tmp)
```

Comment:
If you're interested in more concise ways to construct this list, perhaps look at list comprehensions (i.e., events = [([threading.Event()] * iterations)] * num_threads).